lib/resque.rb

```diff
@@ -539,9 +539,8 @@ def register_hook(name, block)
 
     @hooks ||= {}
     @hooks[name] ||= []
-
-    block = Array(block)
-    @hooks[name].concat(block)
+    @hooks[name] << block
+    return block
```

Comment:
Hmm yeah, I don't think this change really makes sense, hopefully my elaboration on #1074 helps make it easier to reason why.

This change wouldn't completely solve the API incompatibility of the original change, and even if it did I don't believe that it would make semantic sense: the change in #831 fundamentally changed Resque to support a stack of registered worker hooks instead of just one, so it wouldn't make sense for e.g. `Resque.before_fork` to return a single Proc anymore—what Proc would it be, some arbitrary choice of first or last hook in the stack of potentially several?

Here's why this change also incomplete: the change is in `register_hook` so that it indirectly results in its callers now returning a Proc instead of an array, but `register_hook` isn't always called, e.g.

```ruby
def before_fork(&block)
  block ? register_hook(:before_fork, block) : hooks(:before_fork)
end
```

If I call this accessor-style without passing a block argument, the `hooks(:before_fork)` branch is called instead and I still get an array returned, which is still breaking the original method contract.